### ISSUE TITLE
Compiler segfault

### ISSUE THREAD

**Issue by ethernetsellout** [ziglang/zig#19219]
### Zig Version

0.12.0-dev.3161+377ecc6af

### Steps to Reproduce and Observed Behavior

```zig
const std = @import("std");

pub fn range(comptime maximum: comptime_int) struct {
    comptime {
        std.debug.assert(0 < maximum);
    }
} {
    return .{};
}

pub fn main() void {
    const ctx = range(12);
    _ = ctx;
}
```
Output:
`fish: Job 1, 'zig run blah.zig' terminated by signal SIGSEGV (Address boundary error)`

### Expected Behavior

No segfault

**Comment by Rexicon226** [comment]
`0.12.0-dev.3161+377ecc6af` stack trace:
<details>

```
thread 94160 panic: reached unreachable code
Analyzing repro.zig: repro.zig:range__struct_3494.comptime_0
    > %14 = closure_get(%13) node_offset:5:30 to :5:37
      %15 = cmp_lt(@zero, %14) node_offset:5:26 to :5:37
      %16 = break_inline(%12, %15)
    For full context, use the command
      zig ast-check -t repro.zig

  in repro.zig: repro.zig:range__struct_3494.comptime_0
    > %12 = field_call(nodiscard .compile_time, %10, "assert", [
        {%14..%16},
      ]) node_offset:5:9 to :5:38

/home/dr/zig/lib/std/debug.zig:403:14: 0x5c1a6ec in assert (zig)
    if (!ok) unreachable; // assertion failure
             ^
/home/dr/zig/src/Sema.zig:17309:15: 0x6a94e0c in zirClosureGet (zig)
        assert(scope != .none);
              ^
/home/dr/zig/src/Sema.zig:1031:68: 0x65ad71d in analyzeBodyInner (zig)
            .closure_get                  => try sema.zirClosureGet(block, inst),
                                                                   ^
/home/dr/zig/src/Sema.zig:919:30: 0x62bcea7 in analyzeInlineBody (zig)
    if (sema.analyzeBodyInner(block, body)) |_| {
                             ^
/home/dr/zig/src/Sema.zig:945:39: 0x5fd902e in resolveInlineBody (zig)
    return (try sema.analyzeInlineBody(block, body, break_target)) orelse .unreachable_value;
                                      ^
/home/dr/zig/src/Sema.zig:7172:65: 0x71601e7 in analyzeArg (zig)
                const uncoerced_arg = try sema.resolveInlineBody(block, arg_body, zir_call.call_inst);
                                                                ^
/home/dr/zig/src/Sema.zig:7857:56: 0x715e3a9 in analyzeInlineCallArg (zig)
            const casted_arg = try args_info.analyzeArg(ics.caller(), arg_block, arg_i.*, Type.fromInterned(param_ty), func_ty_info, func_inst);
                                                       ^
/home/dr/zig/src/Sema.zig:7572:62: 0x6bfd84a in analyzeCall (zig)
            const opt_noreturn_ref = try analyzeInlineCallArg(
                                                             ^
/home/dr/zig/src/Sema.zig:6883:43: 0x6a9437c in zirCall__anon_111879 (zig)
    const call_inst = try sema.analyzeCall(block, func, func_ty, callee_src, call_src, modifier, ensure_result_used, args_info, call_dbg_node, .call);
                                          ^
/home/dr/zig/src/Sema.zig:1030:62: 0x65ad680 in analyzeBodyInner (zig)
            .field_call                   => try sema.zirCall(block, inst, .field),
                                                             ^
/home/dr/zig/src/Sema.zig:919:30: 0x62bcea7 in analyzeInlineBody (zig)
    if (sema.analyzeBodyInner(block, body)) |_| {
                             ^
/home/dr/zig/src/Sema.zig:945:39: 0x5fd902e in resolveInlineBody (zig)
    return (try sema.analyzeInlineBody(block, body, break_target)) orelse .unreachable_value;
                                      ^
/home/dr/zig/src/Module.zig:3585:50: 0x5fd5657 in semaDecl (zig)
    const result_ref = try sema.resolveInlineBody(&block_scope, decl_bodies.value_body, decl_inst);
                                                 ^
/home/dr/zig/src/Module.zig:3100:32: 0x5d0eb23 in ensureDeclAnalyzed (zig)
        break :blk mod.semaDecl(decl_index) catch |err| switch (err) {
                               ^
/home/dr/zig/src/Compilation.zig:3572:38: 0x5fb21a3 in processOneJob (zig)
            module.ensureDeclAnalyzed(decl_index) catch |err| switch (err) {
                                     ^
/home/dr/zig/src/Compilation.zig:3437:30: 0x5d81f2d in performAllTheWork (zig)
            try processOneJob(comp, work_item, main_progress_node);
                             ^
/home/dr/zig/src/Compilation.zig:2193:31: 0x5d7d62a in update (zig)
    try comp.performAllTheWork(main_progress_node);
                              ^
/home/dr/zig/src/main.zig:4451:24: 0x5dab8d3 in updateModule (zig)
        try comp.update(main_progress_node);
                       ^
/home/dr/zig/src/main.zig:3360:17: 0x5e12f54 in buildOutputType (zig)
    updateModule(comp, color) catch |err| switch (err) {
                ^
/home/dr/zig/src/main.zig:266:31: 0x5c1cc1b in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .run);
                              ^
/home/dr/zig/src/main.zig:204:20: 0x5c199a5 in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/dr/zig/lib/std/start.zig:511:37: 0x5c1942a in main (zig)
            const result = root.main() catch |err| {
                                    ^
../sysdeps/nptl/libc_start_call_main.h:58:16: 0x7f5a5c83c0cf in __libc_start_call_main (../sysdeps/x86/libc-start.c)
../csu/libc-start.c:360:3: 0x7f5a5c83c188 in __libc_start_main_impl (../sysdeps/x86/libc-start.c)
???:?:?: 0x5c190a4 in ??? (???)
???:?:?: 0x0 in ??? (???)
fish: Job 1, 'stage4/bin/zig run repro.zig' terminated by signal SIGABRT (Abort)
```

</details>

**Comment by Rexicon226** [comment]
`zirClosureGet`'s `mod.comptime_capture_scopes.get(key)` is failing to find it because the capture scope index is incorrect. Probably not updating correct in field call.

**Comment by Rexicon226** [comment]
Actually, this issue was fixed in #19190 which merged a couple of hours ago. Can't repro on `0.12.0-dev.3180+83e578a18`.

**Comment by mlugg** [comment]
Yep, that PR eliminated capture scopes altogether, so I guess implicitly solves this bug.
